qiskit_optimization/converters/linear_equality_to_penalty.py

```diff
@@ -59,8 +60,8 @@ def convert(self, problem: QuadraticProgram) -> QuadraticProgram:
         self._src = copy.deepcopy(problem)
         self._dst = QuadraticProgram(name=problem.name)
 
-        # If penalty is None, set the penalty coefficient by _auto_define_penalty()
-        if self._penalty is None:
+        # If no penalty was given, set the penalty coefficient by _auto_define_penalty()
+        if self._should_define_penalty:
```

Comment:
You need to update `self._penalty` after calling `_auto_define_penalty()`.

Reply:
But, wouldn't adding `should_define_penalty` to the constructor create conflict between using `penalty` and calculating a new one? I mean, if I do something like `LinearEqualityToPenalty(penalty=5, should_define_penalty=True)`, what would be the expected behavior?

Reply:
You can make the default of `should_define_penalty` is `True`, and then if `penalty` is given explicitly the `penalty` is used instead of calling `_auto_define_penalty()`.